code/world.dm

```diff
@@ -108,414 +108,60 @@ var/global/datum/global_init/init = new ()
+		response["statuscode"] = 501
+		response["response"] = "Not Implemented"
+		return list2params(response)
 
-		if (bad_key)
-			return bad_key
-
-		var/message = replacetext(input["msg"], "\n", "<br>")
-		world << "<span class=notice><b>[input["announce"] ? input["announce"] : "Administrator"] Announces via Discord:</b><p style='text-indent: 50px'>[message]</p></span>"
-		log_and_message_admins("[input["announce"]] announced remotely: [input["msg"]].")
-
-		return "Announcement successfully sent."
-
-	else if (copytext(T, 1, 8) == "faxlist")
-		var/input[] = params2list(T)
-		var/bad_key = do_topic_spam_protection(addr, input["key"])
-
-		if (bad_key)
-			return bad_key
-
-		if (!ticker)
-			return "Round hasn't started yet! No faxes to display!"
-
-		var/list/faxes = list()
-		switch (input["faxlist"])
-			if ("received")
-				faxes = arrived_faxes
-			if ("sent")
-				faxes = sent_faxes
-
-		if (!faxes || !faxes.len)
-			return "No faxes found!"
-
-		var/list/output = list()
-		for (var/i = 1, i <= faxes.len, i++)
-			var/obj/item/a = faxes[i]
-			output += "ID [i]"
-			output["ID [i]"] = a.name ? a.name : "Untitled Fax"
-
-		return list2params(output)
-
-	else if (copytext(T, 1, 7) == "getfax")
-		var/input[] = params2list(T)
-		var/bad_key = do_topic_spam_protection(addr, input["key"])
-
-		if (bad_key)
-			return bad_key
-
-		var/list/faxes = list()
-		switch (input["received"])
-			if ("received")
-				faxes = arrived_faxes
-			if ("sent")
-				faxes = sent_faxes
-
-		if (!faxes || !faxes.len)
-			return "No faxes found!"
-
-		var/fax_id = text2num(input["getfax"])
-		if (fax_id > faxes.len || fax_id < 1)
-			return "Invalid fax ID!"
-
-		var/output = list()
-		if (istype(faxes[fax_id], /obj/item/weapon/paper))
-			var/obj/item/weapon/paper/a = faxes[fax_id]
-			output["title"] = a.name ? a.name : "Untitled Fax"
-
-			var/content = replacetext(a.info, "<br>", "\n")
-			content = strip_html_properly(content, 0)
-			output["content"] = content
-
-			return list2params(output)
-		else if (istype(faxes[fax_id], /obj/item/weapon/photo))
-			return "The fax is a photo. I cannot send images, unfortunately..."
-		else if (istype(faxes[fax_id], /obj/item/weapon/paper_bundle))
-			var/obj/item/weapon/paper_bundle/b = faxes[fax_id]
-			output["title"] = b.name ? b.name : "Untitled Paper Bundle"
-
-			if (!b.pages || !b.pages.len)
-				return "The bundle was empty! How is that even possible?"
-
-			var/i = 0
-			for (var/obj/item/weapon/paper/c in b.pages)
-				i++
-				var/content = replacetext(c.info, "<br>", "\n")
-				content = strip_html_properly(content, 0)
-				output["content"] += "Page [i]:\n[content]\n\n"
-
-			return list2params(output)
+	if(command.check_params_missing(queryparams))
+		log_debug("API - Mising Params - Status: [command.statuscode] - Response: [command.response]")
+		response["statuscode"] = command.statuscode
+		response["response"] = command.response
+		response["data"] = json_encode(command.data)

```

Comment:
O, yah. Mixing URL encode with JSON encode is a bad idea. Let's go full JSON for return values.